db/migrate/20230508035306_add_type_to_vouchers.rb

```diff
@@ -0,0 +1,12 @@
+class AddTypeToVouchers < ActiveRecord::Migration[7.0]
+  def up
+    add_column :vouchers, :type, :string, limit: 255
+
+    # All the vouchers till now are flat rate voucher
+    Voucher.update_all(type: "Vouchers::FlatRate")
```

Comment:
Sometimes I wonder if it's more efficient to say `default: "Vouchers::FlatRate"` so that the database initialises the value. Uh, that reminds me: should this be `null: false`?

Reply:
Nice one, indeed null should be false. I meant the code guards against it has it would blow up :laughing: but better to enforce it at the database level.